console/readdvd/dvd_udf.cpp

```diff
@@ -729,178 +738,172 @@ static int UDFScanDir( udf_t *udf, struct AD Dir, char *FileName,
+      dvdalign_lbfree(device, directory);
+      return 0;
+    }
+  */
 
-  if( read( udf->dev, directory, lbnum, 2, 0 ) <= 0 ) {
+  if (read(udf->dev, directory, lbnum, 2, 0) <= 0) {
     free(directory);
     return 0;
   }
 
   p = 0;
-  while( p < Dir.Length ) {
-    if( p > DVD_VIDEO_LB_LEN ) {
+  while (p < Dir.Length) {
+    if (p > DVD_VIDEO_LB_LEN) {
       ++lbnum;
       p -= DVD_VIDEO_LB_LEN;
       Dir.Length -= DVD_VIDEO_LB_LEN;
-/*
-      if( DVDReadLBUDF( device, lbnum, 2, directory, 0 ) <= 0 ) {
-        dvdalign_lbfree(device, directory);
+      /*
+            if( DVDReadLBUDF( device, lbnum, 2, directory, 0 ) <= 0 ) {
+              dvdalign_lbfree(device, directory);
+              return 0;
+            }
+      */
+      if (read(udf->dev, directory, lbnum, 2, 0) <= 0) {
+        free(directory);
         return 0;
       }
-*/
-      if( read( udf->dev, directory, lbnum, 2, 0 ) <= 0 ) {
-       free(directory);
-       return 0;
-      }
     }
-    UDFDescriptor( &directory[ p ], &TagID );
-    if( TagID == 257 ) {
-      p += UDFFileIdentifier( &directory[ p ], &filechar,
-                              filename, FileICB );
-      if( !strcasecmp( FileName, filename ) ) {
-        //dvdalign_lbfree(device, directory);
+    UDFDescriptor(&directory[p], &TagID);
+    if (TagID == 257) {
+      p += UDFFileIdentifier(&directory[p], &filechar, filename, FileICB);
+      if (!strcasecmp(FileName, filename)) {
+        // dvdalign_lbfree(device, directory);
         free(directory);
         return 1;
       }
     } else {
-      //dvdalign_lbfree(device, directory);
+      // dvdalign_lbfree(device, directory);
       free(directory);
       return 0;
     }
   }
 
-  //dvdalign_lbfree(device, directory);
+  // dvdalign_lbfree(device, directory);
   free(directory);
   return 0;
 }
 
-static int UDFGetAVDP( udf_t *udf,
-                       struct avdp_t *avdp)
-{
+static int UDFGetAVDP(udf_t *udf, struct avdp_t *avdp) {
   uint8_t *Anchor;
   uint32_t lbnum, MVDS_location, MVDS_length;
   uint16_t TagID;
   uint32_t lastsector;
   int terminate;
-  struct avdp_t; 
-  
-  if(GetUDFCache(udf, AVDPCache, 0, avdp)) {
+  struct avdp_t;
+
+  if (GetUDFCache(udf, AVDPCache, 0, avdp)) {
     return 1;
   }
-  
+
   /* Find Anchor */
   lastsector = 0;
-  lbnum = 256;   /* Try #1, prime anchor */
+  lbnum = 256; /* Try #1, prime anchor */
   terminate = 0;
-  
-  //Anchor = dvdalign_lbmalloc(device, 1);
-  Anchor = (uint8_t*) malloc(2048);
-  if(!Anchor) {
+
+  // Anchor = dvdalign_lbmalloc(device, 1);
+  Anchor = (uint8_t *)malloc(2048);
+  if (!Anchor) {
     return 0;
   }
-  for(;;) {
-    //if( DVDReadLBUDF( device, lbnum, 1, Anchor, 0 ) > 0 ) {
-    if( ! read( udf->dev, Anchor, lbnum, 1, 0 ) ) {
-      UDFDescriptor( Anchor, &TagID );
+  for (;;) {
+    // if( DVDReadLBUDF( device, lbnum, 1, Anchor, 0 ) > 0 ) {
+    if (!read(udf->dev, Anchor, lbnum, 1, 0)) {
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6072101)